src/Cursors.ts

```diff
@@ -137,6 +137,31 @@ export default class Cursors extends EventEmitter<CursorsEventMap> {
     const channel = this.getChannel();
     return await this.cursorHistory.getLastCursorUpdate(channel, this.options.paginationLimit);
   }
+
+  async getSelf(): Promise<CursorUpdate | undefined> {
+    const allCursors = await this.getAll();
+    const self = this.space.getSelf();
+    if (!self || !allCursors) {
+      return undefined;
+    }
+    const cursorUpdates: CursorUpdate[] = Object.values(allCursors).filter(
+      (update): update is CursorUpdate => 'clientId' in update,
+    );
+    return cursorUpdates.find((update) => update.clientId === self.clientId);
+  }
+
+  async getOthers(): Promise<CursorUpdate[]> {
+    const allCursors = await this.getAll();
+    const self = this.space.getSelf();
+    if (!self || !allCursors) {
+      return [];
+    }
+
+    const cursorUpdates: CursorUpdate[] = Object.values(allCursors).filter(
+      (update): update is CursorUpdate => 'clientId' in update,
+    );
+    return cursorUpdates.filter((update) => update.clientId !== self.clientId);
+  }
```

Comment:
As above, think this can be just:

```ts
   const self = this.space.getSelf();
   if (!self) return;

   const allCursors = await this.getAll();
   delete allCursors[self.connectionId];

   return allCursors;
}
```

Note we check for the same `connectionId` too, not `clientId`